document edot java logging (#315)
elastic/opentelemetry: docs/reference/edot-sdks/java/troubleshooting.md

1. Start the application with no agent and see if the issue is not present. Observe if the issue is again present when restarting with the agent.
2. Check end-to-end connectivity without the agent by running one or more of the example apps in [elastic-otel-java](https://github.com/elastic/elastic-otel-java/blob/main/examples/troubleshooting/README.md). These use the OpenTelemetry SDK rather than the auto-instrumentation. They can confirm that the issue is specific to the Java agent or can otherwise identify that the issue is caused by something else.

## Agent debug logging

> Debugging output is verbose and might produce noticeable overhead on the application, there are two main strategies to use:
> - in case of a technical issue or exception with the agent, use [agent debugging](#agent-debugging)
> - if you need details on the captured data, use [per-signal debugging](#per-signal-debugging)

Reviewer: ```suggestion
As debugging output is verbose and might produce noticeable overhead on the application, follow one of these strategies when you need logging:

- In case of a technical issue or exception with the agent, use [agent debugging](#agent-debugging).
- If you need details on the captured data, use [per-signal debugging](#per-signal-debugging).
```

In case of missing data, you should first check that the technology used in the application is supported:
- in [upstream OpenTelemetry Java Instrumentation](https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/main/docs/supported-libraries.md)
- in [EDOT Java](supported-technologies.md)

### Agent debugging

Turning on agent debug logging  

- set `ELASTIC_OTEL_JAVAAGENT_LOG_LEVEL` environment variable (or `elastic.otel.javaagent.log_level` JVM system property) to `debug` (recommended)
- set `OTEL_JAVAAGENT_DEBUG` environment variable (or `otel.javaagent.debug` JVM system property) to `true`

> All of those options require a JVM restart.

Author: Yes, it would make sense to do it this way, do you have any objection @jackshirazi ?

> The `otel.javaagent.debug` / `OTEL_JAVAAGENT_DEBUG` configuration options are inherited from the upstream
> agent, setting them to `true` will also produce span information in plaintext format.
> 
> When using `elastic.otel.javaagent.log_level` / `ELASTIC_OTEL_JAVAAGENT_LOG_LEVEL` set to `debug`, the span information
> will also be included in JSON format, which makes it easier to parse and use for troubleshooting.

Reviewer: ```suggestion
The `otel.javaagent.debug` / `OTEL_JAVAAGENT_DEBUG` configuration options are inherited from the upstream
agent. Setting them to `true` also produce span information in plain text format.

When `elastic.otel.javaagent.log_level` or `ELASTIC_OTEL_JAVAAGENT_LOG_LEVEL` are set to `debug`, the span information is included in JSON format.
```

> In case only the detail of captured data is needed, using [per-signal debugging](#per-signal-debugging) is a more lightweight alternative.

Reviewer: ```suggestion
If only captured data details are needed, [per-signal debugging](#per-signal-debugging) is a lighter alternative.
```

### Per-signal debugging

> Each supported signal can be logged independently, this allows limiting the amount of captured data and reducing the overhead compared

Reviewer: ```suggestion
Each supported signal can be logged independently. This allows limiting the amount of captured data and reducing the overhead compared
```

to [agent debugging](#agent-debugging).

This is configured through the `OTEL_{SIGNAL}_EXPORTER` environment variable or `otel.{signal}.exporter` JVM system property
from the [OpenTelemetry SDK](https://opentelemetry.io/docs/languages/java/configuration/#properties-exporters) by adding any of the following exporters to the default `otlp` value:
- `otlp,logging-otlp`: JSON logging (recommended)
- `otlp,logging`: plain text logging

> All of those options require a JVM restart.

Reviewer: ```suggestion
Both options require a JVM restart.
```
